dev/integration_tests/deferred_components_test/run_release_test.sh

```diff
@@ -32,15 +32,25 @@ java -jar $bundletool_jar_path install-apks --apks=build/app/outputs/bundle/rele
 
 $adb_path shell "
 am start -n io.flutter.integration.deferred_components_test/.MainActivity
-sleep 30
+sleep 20
 exit
 "
-$adb_path logcat -d -t "$script_start_time" > build/app/outputs/bundle/release/run_logcat.log
-echo ""
-if cat build/app/outputs/bundle/release/run_logcat.log | grep -q "Running deferred code"; then
-  echo "All tests passed."
-  exit 0
-fi
-cat build/app/outputs/bundle/release/run_logcat.log
+x=20
+while [ $x -le 120 ]
+do
+  $adb_path logcat -d -t "$script_start_time" > build/app/outputs/bundle/release/run_logcat.log
+  if cat build/app/outputs/bundle/release/run_logcat.log | grep -q "Running deferred code"; then
+    echo "All tests passed."
+    exit 0
+  fi
+  echo "Waiting for $x seconds"
+  # We use an increasing delay here as the emulator can lag and take a long time to finish running the app.
+  # Delay of 20s produces ~5-10% flakes, 30s produces ~2% flakes roughly.
+  $adb_path shell "
+  sleep 10
+  exit
```

Comment:
does it need to run in the emulator? or can it run in the host?

Reply:
Either should be fine. I'll just change to host as it is cleaner.